src/api/controllers/db-controller.ts

```diff
@@ -1111,3 +1112,62 @@ export async function searchTx(
     }
   }
 }
+
+export async function searchHashWithMetadata(
+  hash: string,
+  db: DataStore
+): Promise<FoundOrNot<DbSearchResultWithMetadata>> {
+  // checking for tx
+  const txQuery = await db.getTxListDetails({ txIds: [hash], includeUnanchored: true });
+  if (txQuery.length > 0) {
+    // tx found
+    const tx = txQuery[0];
+    return {
+      found: true,
+      result: {
+        entity_type: 'tx_id',
+        entity_id: tx.tx_id,
+        entity_data: tx,
+      },
+    };
+  }
+  // checking for mempool tx
+  const mempoolTxQuery = await db.getMempoolTxs({
+    txIds: [hash],
+    includeUnanchored: true,
+    includePruned: true,
+  });
+  if (mempoolTxQuery.length > 0) {
+    // mempool tx found
+    const mempoolTx = mempoolTxQuery[0];
+    return {
+      found: true,
+      result: {
+        entity_type: 'mempool_tx_id',
+        entity_id: mempoolTx.tx_id,
+        entity_data: mempoolTx,
+      },
+    };
+  }
+  // checking for block
+  const blockQuery = await db.getBlockWithMetadata({ hash }, { txs: true, microblocks: true });
+  if (blockQuery.found) {
+    // block found
+    const result = parseDbBlock(
```

Comment:
Why not just return the complete `DbBlock` here and let the caller do the parsing? Similar to the other types of results. Seems strange that you're only returning a few things about the block to the user instead of everything we have.

Reply:
Basically, I'm replicating the exact working of the search endpoint. For the case of a block, the endpoint `/extended/v1/search/{id}` responds with `entity_type` as `block_hash`. The client then fetches block data of type `Block` from another endpoint `  /extended/v1/block/{hash}`.
Here, I'm returning the `Block` type block data to the client directly in case it is requested with `include_metadata`.